src/plugins/vis_type_timelion/server/usage_collector/get_usage_collector.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import moment from 'moment';
+import { SearchResponse } from 'elasticsearch';
+import { ElasticsearchClient } from 'src/core/server';
+import { SavedObjectsClientContract, ISavedObjectsRepository } from 'kibana/server';
+import { IFieldType } from '../../../data/common';
+import { extractIndexesFromExpression } from '../../common/utils';
+import { findByValueEmbeddables } from '../../../dashboard/server';
+
+type ESResponse = SearchResponse<{ visualization: { visState: string }; updated_at: string }>;
+
+interface IndexPatternSavedObjectAttrs {
+  title: string;
+  fields: string;
+}
+
+interface VisState {
+  type?: string;
+  params?: any;
+}
+
+export interface TimelionUsage {
+  timelion_use_scripted_fields_90_days_total: number;
+}
+
+const getPastDays = (dateString: string): number => {
+  const date = moment(dateString);
+  return moment().diff(date, 'days');
+};
+
+export const getStats = async (
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract | ISavedObjectsRepository,
+  index: string
+): Promise<TimelionUsage | undefined> => {
+  const timelionUsage = {
+    timelion_use_scripted_fields_90_days_total: 0,
+  };
+
+  const searchParams = {
+    size: 10000,
+    index,
+    ignoreUnavailable: true,
+    filterPath: [
+      'hits.hits._id',
+      'hits.hits._source.visualization',
+      'hits.hits._source.updated_at',
+    ],
+    body: {
+      query: {
+        bool: {
+          filter: { term: { type: 'visualization' } },
+        },
+      },
+    },
+  };
+
+  const { body: esResponse } = await esClient.search<ESResponse>(searchParams);
```

Comment:
NIT: Why not using the `soClient.find` API instead?

Reply:
I used vega usage collector as example. If `soClient.find` is better I can change code. Would you want to change current implementation it?